packages/framework-core/src/booster-auth.ts

```diff
@@ -62,6 +65,66 @@ export class BoosterAuth {
     // Check the roles then
     return userHasSomeRole(user, authorizedRoles)
   }
+
+  public static async verifyToken(config: BoosterConfig, token?: string): Promise<UserEnvelope> {
+    if (!config.tokenVerifier) {
+      return Promise.reject('Token verifier not configured')
+    }
+    if (!token) {
+      return Promise.reject('Empty token')
+    }
+
+    const { issuer, jwksUri } = config.tokenVerifier
+
+    const client = jwksRSA({
+      jwksUri,
+    })
+
+    const verifyOptions: jwt.VerifyOptions = {
+      algorithms: ['RS256'],
+      issuer,
+    }
+
+    return new Promise((resolve, reject) => {
+      const getKey = (header: jwt.JwtHeader, callback: jwt.SigningKeyCallback): void => {
+        if (!header.kid) {
+          callback(new Error('JWT kid not found.'))
+          return
+        }
+
+        client.getSigningKey(header.kid, function(err: any, key: jwksRSA.SigningKey) {
+          if (err) {
+            callback(err, '')
+            return
+          }
+          const signingKey = key.getPublicKey()
+          callback(null, signingKey)
+        })
+      }
+
+      jwt.verify(token, getKey, verifyOptions, (err: any, decoded: any) => {
+        if (err) {
+          reject(err)
+          return
```

Comment:
Isn't this return unnecessary? The promise is already being rejected

Reply:
Agreed!

Reply:
Fixed.